mindmeld/models/text_models.py

```diff
@@ -491,14 +493,134 @@ def _dump(self, path):
 
 
 class PytorchTextModel(PytorchModel):
-    ALLOWED_CLASSIFIER_TYPES = ["embedder", "cnn", "lstm"]
-    pass
+    ALLOWED_CLASSIFIER_TYPES = [v.value for v in SequenceClassificationType.__members__.values()]
+
+    def _get_model_constructor(self):
+        """Returns the class of the actual underlying model"""
+        classifier_type = self.config.model_settings["classifier_type"]
+        embedder_type = self.config.params.get("embedder_type") \
+            if self.config.params is not None else None
+
+        return get_sequence_classifier_cls(
+            classifier_type=classifier_type,
+            embedder_type=embedder_type
+        )
+
+    def evaluate(self, examples, labels):
+        """Evaluates a model against the given examples and labels
+
+        Args:
+            examples: A list of examples to predict
+            labels: A list of expected labels
+
+        Returns:
+            ModelEvaluation: an object containing information about the \
+                evaluation
+        """
+        predictions = self.predict_proba(examples)
+
+        evaluations = [
+            EvaluatedExample(
+                e, labels[i], predictions[i][0], predictions[i][1], self.config.label_type
+            )
+            for i, e in enumerate(examples)
+        ]
+
+        model_eval = StandardModelEvaluation(self.config, evaluations)
+        return model_eval
+
+    def fit(self, examples, labels, params=None):
+
+        if len(set(labels)) <= 1 or not examples:
+            return self
+
+        if not isinstance(examples, PQL.QueryIterator):
+            # pytorch text models are not implemented for role-classifiers, which pass-in an
```

Comment:
I think we should add a check and error message saying NN models are not available for role classifiers if the user selects one in the config. 

I guess this error message won't help the end user identify the problem.

Reply:
Sure, I couldn't find a better spot to raise this error. I will modify the error message written here so that we hint the user about using NNs for role classification.